sentry/src/main/java/io/sentry/transport/AsyncHttpTransport.java

```diff
@@ -163,16 +164,29 @@ public boolean isHealthy() {
 
   @Override
   public void close() throws IOException {
+    close(false);
+  }
+
+  @Override
+  public void close(final boolean isRestarting) throws IOException {
     executor.shutdown();
     options.getLogger().log(SentryLevel.DEBUG, "Shutting down");
     try {
-      if (!executor.awaitTermination(options.getFlushTimeoutMillis(), TimeUnit.MILLISECONDS)) {
+      // We need a small timeout to be able to save to disk any rejected envelope
+      long timeout = isRestarting ? 0 : options.getFlushTimeoutMillis();
+      if (!executor.awaitTermination(timeout, TimeUnit.MILLISECONDS)) {
         options
             .getLogger()
             .log(
                 SentryLevel.WARNING,
-                "Failed to shutdown the async connection async sender within 1 minute. Trying to force it now.");
-        executor.shutdownNow();
+                "Failed to shutdown the async connection async sender  within "
+                    + timeout
+                    + " ms. Trying to force it now.");
+        final @NotNull List<Runnable> runnables = executor.shutdownNow();
+        for (final @NotNull Runnable r : runnables) {
```

Comment:
Is this really required? I thought this is already happening here: https://github.com/getsentry/sentry-java/blob/a537f8a45828b751272940034c4a64ee5f8fd278/sentry/src/main/java/io/sentry/transport/AsyncHttpTransport.java#L134-L146

Reply:
for some reason, the `shutdownNow` method was not always returning the runnable currently executing, so I had to use a field to store the current runnable and use that to store the envelope in case it closes